### PR TITLE
Fix several issues for supporting DPUs

### PR DESCRIPTION
**Pull request by crnithya**
#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: helm/ovn-kubernetes/charts/ovnkube-node-dpu/templates/ovnkube-node-dpu.yaml [hunk 1]
```diff
@@ -229,8 +229,6 @@ spec:
           value: {{ hasKey .Values.global "enableMultiExternalGateway" | ternary .Values.global.enableMultiExternalGateway false | quote }}
         - name: OVNKUBE_NODE_MODE
           value: "dpu"
-        - name: OVNKUBE_NODE_MGMT_PORT_NETDEV
```
Comment (wizhaoredhat) [inline helm/ovn-kubernetes/charts/ovnkube-node-dpu/templates/ovnkube-node-dpu.yaml:232]: Why was this removed?

Comment (crnithya) [reply helm/ovn-kubernetes/charts/ovnkube-node-dpu/templates/ovnkube-node-dpu.yaml:232]: Per https://github.com/ovn-org/ovn-kubernetes/commit/2200972bfb4d1d7554196c7bd98a5620a79817e0, this information will be derived by the DPU from the annotation exposed on the host side.

File: go-controller/pkg/node/default_node_network_controller.go [hunk 2]
```diff
@@ -334,6 +333,12 @@ func setupOVNNode(node *kapi.Node) error {
 		)
 	}
 
+	// In the case of DPU, the hostname should be that of the DPU and not
+	// the K8s Node's. So skip setting the incorrect hostname.
+	if config.OvnKubeNode.Mode != types.NodeModeDPU {
```
[the remark this answers is not in the document]
Comment (crnithya) [reply go-controller/pkg/node/default_node_network_controller.go:338]: this function is called only for non dpu-host modes

[the remark this answers is not in the document]
Comment (cathy-zhou) [reply go-controller/pkg/node/default_node_network_controller.go:338]: I see

**Comment by cathy-zhou** [comment]
lgtm

File: helm/ovn-kubernetes/values-no-ic.yaml [hunk 3]
```diff
@@ -147,6 +147,14 @@ global:
     tag: master
     # -- Image pull policy
     pullPolicy: IfNotPresent
+  # -- Image for DPUs
```
Comment (flavio-fernandes) [inline helm/ovn-kubernetes/values-no-ic.yaml:152]: NIT: Do we also need this in [values-multi-node-zone.yaml](https://github.com/ovn-org/ovn-kubernetes/blob/master/helm/ovn-kubernetes/values-multi-node-zone.yaml) and [values-single-node-zone.yaml](https://github.com/ovn-org/ovn-kubernetes/blob/master/helm/ovn-kubernetes/values-single-node-zone.yaml) ?
https://github.com/ovn-org/ovn-kubernetes/tree/master/helm/ovn-kubernetes

File: helm/ovn-kubernetes/values-no-ic.yaml [hunk 4]
```diff
@@ -138,6 +138,8 @@ global:
   ofctrlWaitBeforeClear: ""
   # -- Separate log file for libovsdb client 
   libovsdbClientLogFile: ""
+  # -- Label for indicating that the node is managing its own network
```
Comment (flavio-fernandes) [inline helm/ovn-kubernetes/values-no-ic.yaml:143]: NIT: Do we also need this in [values-multi-node-zone.yaml](https://github.com/ovn-org/ovn-kubernetes/blob/master/helm/ovn-kubernetes/values-multi-node-zone.yaml) and [values-single-node-zone.yaml](https://github.com/ovn-org/ovn-kubernetes/blob/master/helm/ovn-kubernetes/values-single-node-zone.yaml) ?
https://github.com/ovn-org/ovn-kubernetes/tree/master/helm/ovn-kubernetes

File: helm/ovn-kubernetes/values-no-ic.yaml (hunk 3 above)
Comment (cathy-zhou) [reply helm/ovn-kubernetes/values-no-ic.yaml:152]: I believe this fix does not include dpu support in IC mode.

**Review by girishmg — APPROVED** [review]
LGTM. Evan from Accton has tested this changes as well.

**Comment by wizhaoredhat** [comment]
LGTM. Evan will take a look.

**Comment by EvenChang** [comment]
LGTM as well. I will test again if it is backported to release 1.0.

**Comment by tssurya** [comment]
@crnithya could you please open a backport PR to release-1.0 branch with your commits cherry-picked?

**Comment by crnithya** [comment]
@EvenChang is going to backport this to 1.0 release branch. I will work with him.
